org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/unorderedGroups/GrammarParserTest.java

```diff
@@ -126,6 +131,21 @@ public void setUp() throws Exception {
 		getModelAndExpect("name=(ID & STRING)", 2);
 	}
 	
+	@Test public void testFragments() throws Exception {
+		String grammar =
+			"MyRule;\n" +
+			"MyRule: IntFeature & 'keyword';\n" +
+			"fragment IntFeature: myFeature=INT\n";
+		XtextResource resource = getResourceFromString(grammar);
+		Iterable<EObject> contents = () -> resource.getAllContents();
+		XtextValidator validator = get(XtextValidator.class);
+		boolean result = true;
+		for (final EObject eObject : contents) {
```

Comment:
am, not sure about that @szarnekow 
should validation be done with XtextValidationTest?

Reply:
Sorry, I was on the phone when I provided the code pointers. I might have missed that the validator wasn’t used from one of them. I think adding the parser test is still ok even though it doesn’t fail without the fix. It ensures that it won’t fail in the future, too.